src/lib/y2network/sysconfig/interface_file.rb

```diff
@@ -278,7 +340,14 @@ def save
       #
       # @return [String] Interface's type depending on the file values
       def type
-        "eth"
+        return InterfaceType::DUMMY if @values["INTERFACETYPE"] == "dummy"
+        return InterfaceType::BONDING if defined_variables.any? { |k| k.start_with?("BOND") }
+        return InterfaceType::BRIDGE if defined_variables.any? { |k| k.start_with?("BRIDGE") }
+        return InterfaceType::WIRELESS if defined_variables.any? { |k| k.start_with?("WIRELESS") }
+        return InterfaceType::VLAN if defined_variables.include? ("ETHERDEVICE")
+        return InterfaceType::INFINIBAND if defined_variables.include? ("IPOIB_MODE")
```

Comment:
tun/tap according to mode?

Reply:
Inferring types from values is still missing, and probably part of another card. I added these mainly for testing purpose.